tests/osaka/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -118,6 +122,24 @@ def test_all_opcodes_in_container(
     )
 
 
+@pytest.mark.parametrize(
+    "opcode",
+    sorted(invalid_eof_opcodes | undefined_opcodes),
+)
+def test_invalid_opcodes_after_stop(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test that an invalid opcode placed after STOP (terminating instruction) invalidates EOF.
+    """
+    eof_code = Container(sections=[Section.Code(code=Op.STOP + opcode), Section.Data("00" * 32)])
+    eof_test(
+        data=eof_code,
+        expect_exception=EOFException.UNDEFINED_INSTRUCTION,
```

Comment:
not sure if we want to add this preemptively, but some impls could fail with `UNREACHABLE_CODE` and still make sense. But maybe better to keep simple, it is quite likely that instruction definedness is checked first.

Reply:
It should be possible to define a list of possible exceptions

Reply:
For a combined loop it makes sense to check if the code has been forward referenced before decoding the opcode.  So it could easily be unreachable.

Which begs the question, should we test for all opcodes after a STOP, not just invalid ones?  Is this a code reachability issue or a opcode validity issue?

Reply:
I originally read it as an opcode validity issue, unreachability is added as noise to confuse the eof. But it could be ofc generalized to a reachability issue, with opcode validity as a parameter. It depends on whether and where we have unreachable valid opcodes tested